examples/contrib/oed/ab_test.py

```diff
@@ -0,0 +1,112 @@
+import torch
+import numpy as np
+import matplotlib.pyplot as plt
+
+import pyro
+import pyro.distributions as dist
+from pyro import optim
+from pyro.contrib.oed.eig import vi_ape
+
+
+# Set up regression model dimensions
+N = 100  # number of participants
+p_treatments = 2  # number of treatment groups
+p = p_treatments  # number of features
+prior_stdevs = torch.Tensor([1, .5])
+
+softplus = torch.nn.Softplus()
+
+
+def model(design):
+    # Allow batching of designs
+    loc_shape = list(design.shape)
+    loc_shape[-2] = 1
+    loc = torch.zeros(loc_shape)
+    scale = prior_stdevs
+    # Place a normal prior on the regression coefficient
+    # w is 1 x p: hence use .independent(2)
+    w_prior = dist.Normal(loc, scale).independent(2)
+    w = pyro.sample('w', w_prior).transpose(-1, -2)
+
+    with pyro.iarange("map", N, dim=-2):
+        # Run the regressor forward conditioned on inputs
+        prediction_mean = torch.matmul(design, w).squeeze(-1)
+        # y is an n-vector: hence use .independent(1)
+        pyro.sample("y", dist.Normal(prediction_mean, 1).independent(1))
+
+
+def guide(design):
+    loc_shape = list(design.shape)
+    loc_shape[-2] = 1
+    # define our variational parameters
+    w_loc = torch.zeros(loc_shape)
+    # note that we initialize our scales to be pretty narrow
+    w_sig = -3*torch.ones(loc_shape)
+    # register learnable params in the param store
+    mw_param = pyro.param("guide_mean_weight", w_loc)
+    sw_param = softplus(pyro.param("guide_scale_weight", w_sig))
```

Comment:
Prefer `torch.distribution.constraints`, e.g.
```
from torch.distributions import constraints
...
sw_param = pyro.param("guide_scale_weight", w_sig, constraint=constraints.positive)
```

Reply:
Curiously, this didn't work (led to `nan`s). The authors of the Bayesian regression tutorial may have insight (it's copied from there). I'll leave as `torch.nn.functional.softplus` for now

Reply:
I think the issue here was that you'd have to pass an initial positive value for the param, say `w_sig = torch.ones(loc_shape)`; otherwise it'll assume that the initial value that is passed is already constrained and try to invert the transformation to get the unconstrained value (i.e. applying log on a negative value in this case which returns nans). `softplus` is fine though, and we can update both this and the bayesian regression example later.

Reply:
Hmm I did switch the initial val to `1e-3` for that very reason, still not joy.